examples/platform/nxp/se05x/DeviceAttestationSe05xCredsExample_v2.cpp

```diff
@@ -178,108 +182,78 @@ CHIP_ERROR ExampleSe05xDACProviderv2::SignWithDeviceAttestationKey(const ByteSpa
 
     msg_reader.Init(message_to_sign);
 
-    /* To be removed. Use common key id to sign message */
-    static bool sign_cert_decl_attest = 1;
+    uint16_t taglen = 0;
+    ByteSpan tagvalue;
+    uint8_t tempBuf[2] = {
+        0,
+    };
+
 
-    if (sign_cert_decl_attest)
+    tempBuf[0] = (uint8_t) TLV::TLVElementType::Structure;
+    SuccessOrExit(se05xSetCertificate(START_CONTAINER_SE05X_ID, tempBuf, 1));
+
+    for (int i = 1; i <= 3; i++)
     {
-        /* Check if certificate declaration tag is present and Skip certificate declaration tag */
-        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(1), tagReader));
-
-        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(2), tagReader));
-        uint8_t attlen = tagReader.GetLength();
-        VerifyOrReturnError(attlen > 0, CHIP_ERROR_INVALID_TLV_TAG);
-        /* Get attestation nonce */
-        ByteSpan attest_nonce;
-        ReturnErrorOnFailure(tagReader.Get(attest_nonce));
-        /* Set attestation nonce */
-        VerifyOrReturnError(CHIP_NO_ERROR ==
-                                se05xSetCertificate(CD_ATTEST_NONCE_DATA_SE05X_ID, attest_nonce.data(), attest_nonce.size()),
-                            CHIP_ERROR_INTERNAL);
-
-        ReturnErrorOnFailure(TLV::Utilities::Find(msg_reader, TLV::ContextTag(3), tagReader));
-        uint8_t tslen = tagReader.GetLength();
-        if (tslen > 0)
+        CHIP_ERROR tlverr = CHIP_NO_ERROR;
+        tlverr = TLV::Utilities::Find(msg_reader, TLV::ContextTag(i), tagReader);
+        if (tlverr == CHIP_ERROR_TLV_TAG_NOT_FOUND){
+            continue;
+        }
+        SuccessOrExit(tlverr);
+
+        taglen     = tagReader.GetLength();
+        tempBuf[0] = tagReader.GetControlByte();
+        ;
+        tempBuf[1] = i;
+        SuccessOrExit(se05xSetCertificate(TAG1_ID + (3 * (i - 1)), tempBuf, 2));
```

Comment:
Where is this `3 * (i - 1)` bit coming from?  Should at least be documented, if not given a nice macro or function.

Reply:
Updated the function with the comments on this logic.